consensus/state_test.go

```diff
@@ -183,7 +184,9 @@ func TestStateEnterProposeYesPrivValidator(t *testing.T) {
 	ensureNoNewTimeout(timeoutCh, cs.config.TimeoutPropose.Nanoseconds())
 }
 
+// This test injects invalid field to block and checks if state discards it by voting nil
 func TestStateBadProposal(t *testing.T) {
+	t.Skip("Block Executor don't have any validation for types.Data fields and we can't inject bad data there")
```

Comment:
Pointing to this. Currently don't have ideas on what validation rules to add to block exec for types.Data specifically.